### PR TITLE
History

### PR DESCRIPTION
**Pull request by MitchellWT**
This MR does the following:
- Adds database storage/support.
  - Still provides standard operations when database is not available.
- Adds container compatibility for database.
- Adds basic unit testing for storage manager.

### CODE REVIEW

File: gui/src/components/Suites.js [hunk 1]
```diff
@@ -25,15 +26,9 @@ function Suites({socket}) {
...
+  }, [suites, setSuite]);
   
-  socket.onmessage = (event) => {
-    const data = event.data;
-    const jsonData = JSON.parse(data);
-    console.log(`Recived message ${data}`);
-    addAndUpdateSuites(suites, jsonData);
-    console.log(suites);
-  };
+  useEffect(() => addAndUpdateSuites(suiteMessage), [suiteMessage]);
```
Comment (Oblivion-21) [inline gui/src/components/Suites.js:31]: Getting an error from this line 
`front_end      | [0]   Line 31:53:  React Hook useEffect has a missing dependency: 'addAndUpdateSuites'. Either include it or remove the dependency array  react-hooks/exhaustive-deps`

File: gui/src/components/History.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    if (socket.readyState === 1) socket.send(msg);
+  };
+
+  return (
+    <Container fluid className="p-5 d-grid gap-4">
+      <Button onClick={updateHistory}>Update History</Button>
+      <Accordion>
+        {history === undefined ? <></> : history.map((suite, index) => (
+          <Accordion.Item eventKey={index} key={index}>
+            <Accordion.Header>Suite {suite["suiteID"]} | {suite["url"]}</Accordion.Header>
```
Comment (Oblivion-21) [inline gui/src/components/History.js:27]: I think calling them Suite N in the UI is kinda confusing but maybe that just me

File: gui/src/components/Suites.js (hunk 1 above)
Comment (Oblivion-21) [inline gui/src/components/Suites.js:31]: Getting an error from this line
`front_end | [0] Line 31:53: React Hook useEffect has a missing dependency: 'addAndUpdateSuites'. Either include it or remove the dependency array react-hooks/exhaustive-deps`

**Review by Oblivion-21 — CHANGES_REQUESTED** [review]
Awesome work on the UI updates, they are suite

File: gui/src/components/History.js (hunk 2 above)
Comment (MitchellWT) [reply gui/src/components/History.js]: Fixed

**Pull request merged**